qiskit/quantum_info/operators/base_operator.py

```diff
@@ -97,11 +97,6 @@ def _output_dim(self):
         """Return the total input dimension."""
         return self._op_shape._dim_l
 
-    @property
-    def settings(self):
-        """Return operator settings."""
-        return {"op_shape": self._op_shape}
-
```

Comment:
Is this part of the ABC interface for everything that inherits from `BaseOperator`?  If so, we could add it as an abstract method.

Reply:
It currently is, but I don't think it needs to be. It's really API for serializable objects for runtime than for operators themselves, so for now I think it's easiest to just add the settings to the specific operators that are actually used in runtime or qiskit-experiments.

Reply:
I'm not super wild about a magic duck-typed method being called something as generic as `settings`, but that's a pre-existing problem, not the responsibility of this PR.  If you're happy that the presence of the method is sufficient to indicate to a serialiser that it can be serialised, I'm happy to approve the PR as-is.

Reply:
Yeah I could see a less generic name being better, but that is something you will have to take up with the ibmq provider / runtime folks. I think this is the best option given the current context.